airbyte-webapp/src/views/Connection/ConnectionForm/components/ScheduleField.tsx

```diff
@@ -0,0 +1,182 @@
+    }
+
+    return currentSelectedZone;
+  };
+
+  const onCronInputChange = (
+    event: ChangeEvent<HTMLInputElement>,
+    field: FieldInputProps<ConnectionScheduleDataCron>
+  ) => {
+    setFieldValue(field.name, {
+      ...field.value,
+      cronExpression: event.currentTarget.value,
+    });
+  };
+
+  const onCronZoneChange = (event: DropDownRow.IDataItem, field: FieldInputProps<ConnectionScheduleDataCron>) => {
+    setFieldValue(field.name, {
+      ...field.value,
+      cronTimeZone: event.value,
+    });
+  };
+
+  const cronTimeZones = useMemo(() => {
+    return availableCronTimeZones.map((zone: string) => ({ label: zone, value: zone }));
+  }, []);
+
+  return (
+    <>
+      <Field name="scheduleData.basicSchedule">
+        {({ field, meta }: FieldProps<ConnectionScheduleDataBasicSchedule>) => (
+          <div className={styles.flexRow}>
+            <div className={styles.leftFieldCol}>
+              <ControlLabels
+                className={styles.connectorLabel}
+                nextLine
+                error={!!meta.error && meta.touched}
+                label={formatMessage({
+                  id: "form.frequency",
+                })}
+                message={formatMessage({
+                  id: "form.frequency.message",
+                })}
+              />
+            </div>
+            <div className={styles.rightFieldCol} style={{ pointerEvents: mode === "readonly" ? "none" : "auto" }}>
+              <DropDown
+                {...field}
+                error={!!meta.error && meta.touched}
+                options={frequencies}
+                onChange={(item) => {
+                  onScheduleChange(item, field);
+                }}
+                value={getBasicScheduleValue(field.value)}
+              />
+            </div>
+          </div>
+        )}
+      </Field>
+      {isCustomSchedule && (
+        <Field name="scheduleData.cron">
+          {({ field, meta }: FieldProps<ConnectionScheduleDataCron>) => (
+            <div className={styles.flexRow}>
+              <div className={styles.leftFieldCol}>
+                <ControlLabels
+                  className={styles.connectorLabel}
+                  nextLine
+                  error={!!meta.error && meta.touched}
+                  label={formatMessage({
+                    id: "form.cronExpression",
+                  })}
+                />
+              </div>
+              <div className={styles.rightFieldCol} style={{ pointerEvents: mode === "readonly" ? "none" : "auto" }}>
+                <div className={styles.flexRow}>
+                  <Input
+                    disabled={mode === "readonly"}
+                    error={!!meta.error}
+                    data-testid="cronExpression"
+                    onBlur={field.onBlur}
+                    placeholder={formatMessage({
+                      id: "form.cronExpression.placeholder",
+                    })}
+                    value={field.value?.cronExpression}
```

Comment:
Do we do any validations on cron expressions? I entered a cron string that I thought was valid but it turned out it wasn't, and the UI only said "form is invalid" at the bottom with no indication that it was the cron string that was invalid. 

It would be helpful to improve the UX around this, though I can also understand if that is a larger effort that should be done in a separate PR.

Reply:
Yeah I also feel the same we need to improve the error messages and experience there. Can pick in the next PR